<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RoomCreationWidgetClass</class>
 <widget class="QWidget" name="RoomCreationWidgetClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>627</width>
    <height>454</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RoomCreationWidget</string>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>40</number>
   </property>
   <property name="topMargin">
    <number>40</number>
   </property>
   <property name="rightMargin">
    <number>40</number>
   </property>
   <property name="bottomMargin">
    <number>40</number>
   </property>
   <item row="12" column="1">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="8" column="1">
    <widget class="QLabel" name="roomPswdErrorLabel">
     <property name="styleSheet">
      <string notr="true">color:red;</string>
     </property>
     <property name="text">
      <string>pswd error</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,1">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Your room name</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="roomNameLineEdit"/>
     </item>
    </layout>
   </item>
   <item row="0" column="1">
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="6" column="1">
    <widget class="QLabel" name="roomNameErrorLabel">
     <property name="styleSheet">
      <string notr="true">color:red;</string>
     </property>
     <property name="text">
      <string>name err</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,1">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Your room password</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="pswdLineEdit"/>
     </item>
    </layout>
   </item>
   <item row="10" column="0">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="9" column="1">
    <spacer name="verticalSpacer_3">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="10" column="1">
    <widget class="QPushButton" name="createRoomButton">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>250</width>
       <height>50</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>430</width>
       <height>56</height>
      </size>
     </property>
     <property name="baseSize">
      <size>
       <width>430</width>
       <height>65</height>
      </size>
     </property>
     <property name="text">
      <string>create room</string>
     </property>
    </widget>
   </item>
   <item row="10" column="3">
    <spacer name="horizontalSpacer_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="11" column="1">
    <widget class="QPushButton" name="backToMenuButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>250</width>
       <height>50</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>430</width>
       <height>65</height>
      </size>
     </property>
     <property name="text">
      <string>back to menu</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>backToMenuButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>backToMenuButton</sender>
   <signal>clicked()</signal>
   <receiver>RoomCreationWidgetClass</receiver>
   <slot>emitBackToMenu_slot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>319</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>206</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>backToMenuButton</sender>
   <signal>clicked()</signal>
   <receiver>RoomCreationWidgetClass</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>87</x>
     <y>362</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>createRoomButton</sender>
   <signal>clicked()</signal>
   <receiver>RoomCreationWidgetClass</receiver>
   <slot>createRoomBtnPressed_slot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>438</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>337</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>roomNameLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>RoomCreationWidgetClass</receiver>
   <slot>validateFields_slot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>343</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>489</x>
     <y>73</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pswdLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>RoomCreationWidgetClass</receiver>
   <slot>validateFields_slot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>validateFields_slot()</slot>
  <slot>createRoomBtnPressed_slot()</slot>
  <slot>tryJoining_slot()</slot>
 </slots>
</ui>
